corpus/filter.py

```diff
@@ -176,12 +180,21 @@ def run(self):
         segments = set(segments)
         c = corpus.Corpus()
         c.load(tk.uncached_path(self.bliss_corpus))
-        for rec in c.all_recordings():
+
+        to_delete = []
+        for idx, rec in enumerate(c.all_recordings()):
             if self.invert_match:
                 rec.segments = [x for x in rec.segments if x.fullname() not in segments and x.name not in segments]
             else:
                 rec.segments = [x for x in rec.segments if x.fullname() in segments or x.name in segments]
 
+            if not rec.segments:
+                to_delete.append(idx)
+
+        if self.delete_empty_recordings:
+            for idx in reversed(to_delete):
+                del c.subcorpora[0].recordings[idx]
```

Comment:
this does only work for the specific corpus structure that we use, but not in general.

I would suggest you use the function [Corpus.remove_recording](https://github.com/rwth-i6/i6_core/blob/db0f0a3b17419b221fdecdf5f8a996ed6b8451e6/lib/corpus.py#L188)

e.g. 
```
if not rec.segments:
  to_delete.append(rec)

[..]
for rec in to_delete:
  c.remove_recording(rec)
```
and then remove the `idx` and `enumerate`
(not tested)

Reply:
Thanks for the comment!
I used `remove_recording` like your snippet (also changed to pass a Recording obj and not idx). 
And run a test on a `/nas/data/speech/FR_FR/16kHz/commonvoice/corpus/other.v1/commonvoice-other.v1.xml.gz` corpus, which seems to work.